data/helpers.d/setting

```diff
@@ -178,25 +187,62 @@ ynh_webpath_register () {
 #
 # Requires YunoHost version 3.7.0 or higher.
 ynh_permission_create() {
-	# Declare an array to define the options of this helper.
-	local legacy_args=pua
-    declare -Ar args_array=( [p]=permission= [u]=url= [a]=allowed= )
+    # Declare an array to define the options of this helper.
+    local legacy_args=puAhaltP
+    declare -Ar args_array=( [p]=permission= [u]=url= [A]=additional_urls= [h]=auth_header= [a]=allowed= [l]=label= [t]=show_tile= [P]=protected= )
     local permission
     local url
+    local additional_urls
+    local auth_header
     local allowed
+    local label
+    local show_tile
+    local protected
     ynh_handle_getopts_args "$@"
 
     if [[ -n ${url:-} ]]; then
-        url="'$url'"
-    else
-        url="None"
+        url=",url='$url'"
+    fi
+
+    if [[ -n ${additional_urls:-} ]]; then
+        additional_urls=",additional_urls=['${additional_urls//';'/"','"}']"
+    fi
+
+    if [[ -n ${auth_header:-} ]]; then
+        if [ $auth_header == "true" ]; then
+            auth_header=",auth_header=True"
+        else
+            auth_header=",auth_header=False"
+        fi
     fi
 
     if [[ -n ${allowed:-} ]]; then
         allowed=",allowed=['${allowed//';'/"','"}']"
     fi
+
+    if [[ -n ${label:-} ]]; then
+        label=",label='$label'"
+    else
+        label=",label='$YNH_APP_LABEL ($permission)'"
+    fi
+
+    if [[ -n ${show_tile:-} ]]; then
+        if [ $show_tile == "true" ]; then
+            show_tile=",show_tile=True"
+        else
+            show_tile=",show_tile=False"
+        fi
+    fi
+
+    if [[ -n ${protected:-} ]]; then
+        if [ $protected == "true" ]; then
+            protected=",protected=True"
+        else
+            protected=",protected=False"
+        fi
+    fi
     
-    yunohost tools shell -c "from yunohost.permission import permission_create; permission_create('$app.$permission', url=$url ${allowed:-} , sync_perm=False)"
+    yunohost tools shell -c "from yunohost.permission import permission_create; permission_create('$app.$permission' ${url:-} ${additional_urls:-} ${auth_header:-} ${allowed:-} ${label:-} ${show_tile:-} ${protected:-} , sync_perm=False)"
```

Comment:
Helpers are **shell** functions.
To maintain those helpers, helpers are to stay **bash** functions.

A yunohost command should be used here.

Reply:
If I remember correctly, it was designed this way to avoid making the permission creationg/deletion/update/.. API/CLI public because we want to avoid users messing with permissions manually ... But clearly that helper code is becoming quite funky.

Maybe a middle-ground to have a special helper that's able to call yunohost functions directly such that we can focus on the bash logic inside the actual helpers but idk how realistic or better this would be.

Reply:
Yet users should be able to deal with permission on their own if they want to.
Especially considering a CLI tool will refrain crazy usage.

What's for sure is that python command inside a bash helper can be maintain easily.

Reply:
>Yet users should be able to deal with permission on their own if they want to.

Yes if that's about updating the permission like adding/removing users, but here's we're talking about things like creating/deleting permission which can make everything explode if you start deleting/creating "main" permissions yourself ... Or any permission in fact ... unless you know exactly what you're doing ...

Reply:
Didn't go througt that permission enough to say if needed or not.
But I feel like sooner or later, we'll want to have a CLI tool for that.

Anyway, that's not really the problem here.
The problem is that bash helper should not call python directly into their code.

Reply:
Oh damn I missed that !!!
Especially when that python code is actually full of bash substitutions !